### ISSUE TITLE
check for conditions with systematic results

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#70]
Originally reported by: **Anonymous**

---

Good day,

i thought that pylint might check for conditions predictably evaluating as always True or False.
It should be easy to detect slices equality with obvious different lengths (which should report as a coding error), but there are definitively other cases (equality check on different standard types for example) where a warning might be raised to the attention of the code reviewer.

Conditions always evaluating the same way are most probably coding error or obfuscation.

```
#!python

# easy to detect
always_true = line[:4] != 'From '
always_false = time.localtime()[:3] == (2013, 10, 12, 1)

# harder
s = slice(0, 5, 2)
x = [0, 2]
always_false = range(100)[s] == x

# way harder: predictable different types
y = 0, 2
always_false = x == y
```

Regards,
Romain

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/70

**Comment by PCManticore** [comment]
I find this too difficult to be worth it (e.g too many man-hour spent on implementing such a feature)
